blockstore/apps/api/v1/views/snapshots.py

```diff
@@ -0,0 +1,105 @@
+"""
+Views for Snapshots.
+"""
+
+from django import http
+from rest_framework import status
+from rest_framework import viewsets
+from rest_framework.response import Response
+
+from blockstore.apps.bundles.store import BundleDataStore
+from blockstore.apps.bundles.models import BundleVersion
+
+from ...constants import FILE_PATH_REGEX
+from ..serializers.snapshots import FileInfoSerializer
+
+
+class BundleFileReadOnlyViewSet(viewsets.ViewSet):
+    """
+    Read only ViewSet for Snapshot files.
+    """
+
+    detail_view_name = 'api:v1:bundleversionfile-detail'
+
+    lookup_field = 'path'
+    lookup_value_regex = FILE_PATH_REGEX
+
+    serializer_class = FileInfoSerializer
+
+    def get_bundle_version(self, bundle_uuid, version_num=None):
+
+        filter_kwargs = {
+            'bundle__uuid': bundle_uuid
+        }
+        if version_num:
+            filter_kwargs['version_num'] = version_num
+
+        try:
+            return BundleVersion.objects.filter(**filter_kwargs).order_by('-version_num').first()
+        except BundleVersion.DoesNotExist:
+            pass
+        return None
+
+    def get_serializer_context(self):
+        return {
+            'detail_view_name': self.detail_view_name,
+            'format': self.format_kwarg,
+            'request': self.request,
+        }
+
+    def list(self, _request, bundle_uuid, version_num=None):
+        response_data = []
+        bundle_version = self.get_bundle_version(bundle_uuid, version_num)
+        if bundle_version:
+            snapshot = bundle_version.snapshot()
+            file_serializer = FileInfoSerializer(
+                snapshot.files.values(),
+                context=self.get_serializer_context(),
+                many=True
+            )
+            response_data = file_serializer.data
+        return Response(response_data)
+
+    def retrieve(self, _request, bundle_uuid, version_num=None, path=None):
+        response_data = []
+        bundle_version = self.get_bundle_version(bundle_uuid, version_num)
+        if bundle_version:
+            snapshot = bundle_version.snapshot()
+            file = snapshot.files.get(path, None)
+            if not file:
+                raise http.Http404
+
+            file_serializer = FileInfoSerializer(
+                file,
+                context=self.get_serializer_context(),
+            )
+            response_data = file_serializer.data
+        return Response(response_data)
```

Comment:
Lots of duplicated code between `list` and `retrieve`.. can these be refactored without being confusing?

Reply:
Moved some of the common code to `get_bundle_version_or_404()`.

Reply:
Lovely 👍